### PR TITLE
Fix inversion of port label when loading a topology

### LINKED ISSUE #954: Multiple links between devices and the interface labels shifts
See https://gns3.com/qa/bug-multiple-links-between-devic

Tested it with GNS3 version 1.4.0 on Darwin (64-bit) with Python 3.5.1 Qt 5.5.1.
* create new empty project
* add two 3725 router
* connect both interface of R1 to R2
* rearrange interface labels
* save project<br>test1
* reopen project<br>test2

The position of the interface labels of fa0/0 are correct, but the interface labels of fa0/1 are on a different position after a re-open.

### PR DESCRIPTION
**Pull request by noplay**
This PR rewrite the way multi link are created and avoid
to invert their directions.

Fix #954

### CODE REVIEW

File: gns3/graphics_view.py [hunk 1]
```diff
@@ -293,36 +294,39 @@ def addLinkSlot(self, link_id):
...
-                source_port, destination_port = destination_port, source_port
-                source_item, destination_item = destination_item, source_item
-                multi = d1 + 1
-            elif d1 >= d2:
-                source_port, destination_port = destination_port, source_port
-                source_item, destination_item = destination_item, source_item
-                multi = d2
-            else:
-                multi = d1
+        # Multi-link management
```
Comment (grossmj) [inline gns3/graphics_view.py]: Great drawing! :+1:

File: gns3/graphics_view.py [hunk 2]
```diff
@@ -293,36 +294,39 @@ def addLinkSlot(self, link_id):
...
+        # MAX 7 links on the scene between 2 nodes
+        if multi > 7:
             multi = 0
+        # Pair item represent the bottom links
+        elif multi % 2 == 0:
+            multi = multi / 2
+        else:
+            multi = math.ceil(float(multi) / 2) * -1
+
+        print(multi)
```
Comment (grossmj) [inline gns3/graphics_view.py]: Debug ;)

File: gns3/graphics_view.py [hunk 3]
```diff
@@ -293,36 +294,39 @@ def addLinkSlot(self, link_id):
...
+        # Multi-link management
+        #
+        # multi is the offset of the link
+        # +------+       multi = -1    Link 3  +-------+
+        # |      +-----------------------------+       |
+        # |  R1  |                             |   R2  |
+        # |      |        multi = 0    Link 1  |       |
+        # |      +-----------------------------+       |
+        # |      |        multi = 1    Link 2  |       |
+        # +------+-----------------------------+-------+
```
Comment (ehlers) [inline gns3/graphics_view.py]: My tests show slightly different values: link 1 = multi 0, link2 = multi -1, link3 = multi 1, link 4 = multi -2, link 5 = multi 2, ...

File: gns3/graphics_view.py [hunk 4]
```diff
@@ -293,36 +294,39 @@ def addLinkSlot(self, link_id):
...
 
-        if source_item == destination_item:
+        # MAX 7 links on the scene between 2 nodes
+        if multi > 7:
             multi = 0
+        # Pair item represent the bottom links
+        elif multi % 2 == 0:
+            multi = multi / 2
+        else:
+            multi = math.ceil(float(multi) / 2) * -1
```
Comment (ehlers) [inline gns3/graphics_view.py]: Floating point is evil :sunglasses:

I would change line 324 to "multi = multi // 2", that way it stays integer

For line 326: These lines are equivalent:
multi = math.ceil(float(multi) / 2) * -1
multi = math.floor((-1 * float(multi)) / 2)
multi = (-1* multi) // 2
multi = -multi // 2

**Comment by ehlers** [comment]
Made some tests and found it working well, nice work.

File: gns3/graphics_view.py (hunk 2 above)
Comment (noplay) [inline gns3/graphics_view.py]: Removed

File: gns3/graphics_view.py (hunk 3 above)
Comment (noplay) [inline gns3/graphics_view.py]: You are right doc is not sync with final implementation

File: gns3/graphics_view.py (hunk 4 above)
Comment (noplay) [inline gns3/graphics_view.py]: A 324 it's not possible to get a float. But you are right i will rewrote it with your suggest

**Comment by noplay** [comment]
Thanks for the feedbacks

**Pull request merged**